support-lib/objc/DJIMarshal+Private.h

```diff
@@ -29,7 +29,7 @@ struct Bool {
 
     struct Boxed {
         using ObjcType = NSNumber*;
-        static CppType toCpp(ObjcType x) noexcept { assert(x); return Bool::toCpp([x boolValue]); }
+        static CppType toCpp(ObjcType x) noexcept { assert(x != nil); return Bool::toCpp([x boolValue]); }
```

Comment:
Interesting.  Stylistically I generally do prefer to let pointers be treated as bools directly, but in this context I'm fine with changing the code to be more generally usable by people who don't like that style and turn on warnings against it.